### PR TITLE
Add support for changing the loopback flag

### PR DESCRIPTION
**Pull request by jjguti**
Small PR to allow the local loopback to be disabled/enabled on a socketcan interface.

### CODE REVIEW

**Comment by felixdivo** [comment]
@jjguti This looks like a neat addition. But I think we should add a test to make sure it correctly loops back and only does so if the flag is set. Could You add that? Then, this can be merged.

**Comment by jjguti** [comment]
@felixdivo Hi, I have now added a test for the local_loopback flag, please tell me if it's good enough for a merge or if something else should be added.

**Comment by felixdivo** [comment]
Thanks for your quick reply! And sorry for the long pause, there is recently quite a backlog of issues/PRs.

I think that the test is already nearly complete, but we should add a few lines, see my comments below.

File: test/test_socketcan_loopback.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        on the local device"""
+        loopback_send_bus = can.Bus(
+            interface=self.INTERFACE_1,
+            channel=self.CHANNEL_1,
+            bitrate=self.BITRATE,
+            local_loopback=True,
+        )
+        try:
+            msg = can.Message(arbitration_id=0x123, is_extended_id=False)
+            loopback_send_bus.send(msg)
```
Comment (felixdivo) [inline test/test_socketcan_loopback.py:45]: Test that `loopback_send_bus.recv()` returns a message.

File: test/test_socketcan_loopback.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        noloopback_send_bus = can.Bus(
+            interface=self.INTERFACE_1,
+            channel=self.CHANNEL_1,
+            bitrate=self.BITRATE,
+            local_loopback=False,
+        )
+        try:
+            msg = can.Message(arbitration_id=0x123, is_extended_id=False)
+            noloopback_send_bus.send(msg)
+            recv_msg = self._recv_bus.recv(self.TIMEOUT)
```
Comment (felixdivo) [inline test/test_socketcan_loopback.py:63]: Test that `Test that `loopback_send_bus.recv()` returns `None`.

File: test/test_socketcan_loopback.py (hunk 1 above)
Comment (jjguti) [reply test/test_socketcan_loopback.py:45]: loopback_send_bus.recv() does NOT return a message. The local_loopback flag (as far as I understand it), means the *other* local sockets open on the same machine will receive the message, but not your own socket. I can instead add an assert that it does not recv() in any case.

Comment (jjguti) [reply test/test_socketcan_loopback.py:45]: A bit more info can be read here:

https://www.kernel.org/doc/html/latest/networking/can.html#socketcan-local-loopback1

Comment (felixdivo) [reply test/test_socketcan_loopback.py:45]: Whoops. OK, that also sounds useful (even more useful). Then yes, do as you said. Maybe briefly document it in `SocketcanBus`?

Comment (jjguti) [reply test/test_socketcan_loopback.py:45]: Added. This was very confusing for me also when I started looking at it :)

**Review by hardbyte — APPROVED** [review]
👍🏼

**Comment by felixdivo** [comment]
Nice @jjguti!

**Review by felixdivo — APPROVED** [review]

**Pull request merged**